### ISSUE TITLE
bug: cannot set project to Public after setting it to Restricted

### ISSUE THREAD

**Issue by Rexogamer** [WeblateOrg/weblate#9416]
### Describe the issue

I'm an admin on a self-hosted instance of Weblate (https://weblate.insrt.uk). I accidentally set a project to Restricted - however, when I went to set it back to Public I found I couldn't? I can set it to Private or Custom, then set it back to Restricted - however, when I try to set it to Public, it doesn't seem to save properly (it claims that the change was saved, but going back to the Access tab reveals it wasn't).

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

1. Go to a project's settings
2. Open the Access tab
3. Set it to Restricted (from Public) and click Save
4. Go back to the Access tab after the page reloads
5. Observe that the change hasn't been saved

### Expected behavior

The change should be saved and the project should be set to Public.

### Screenshots

[Screencast from 2023-06-14 23-12-46.webm](https://github.com/WeblateOrg/weblate/assets/42586271/ac6ae3e8-79fd-433d-8d4c-2c2ea926023d)

### How do you run Weblate?

Docker container

### Weblate versions

 * Weblate: 4.17
 * Django: 4.2.1
 * siphashc: 2.1
 * translate-toolkit: 3.8.6
 * lxml: 4.9.2
 * Pillow: 9.5.0
 * nh3: 0.2.12
 * python-dateutil: 2.8.2
 * social-auth-core: 4.4.2
 * social-auth-app-django: 5.2.0
 * django-crispy-forms: 2.0
 * oauthlib: 3.2.2
 * django-compressor: 4.3.1
 * djangorestframework: 3.14.0
 * django-filter: 23.1
 * django-appconf: 1.0.5
 * user-agents: 2.2.0
 * filelock: 3.11.0
 * rapidfuzz: 3.0.0
 * openpyxl: 3.1.2
 * celery: 5.2.7
 * django-celery-beat: 2.5.0
 * kombu: 5.3.0
 * translation-finder: 2.15
 * weblate-language-data: 2023.4
 * html2text: 2020.1.16
 * pycairo: 1.23.0
 * pygobject: 3.44.1
 * diff-match-patch: 20200713
 * requests: 2.28.2
 * django-redis: 5.2.0
 * hiredis: 2.2.3
 * sentry_sdk: 1.25.0
 * Cython: 0.29.35
 * misaka: 2.1.1
 * GitPython: 3.1.31
 * borgbackup: 1.2.4
 * pyparsing: 3.0.9
 * pyahocorasick: 2.0.0
 * python-redis-lock: 4.0.0
 * charset-normalizer: 3.1.0
 * Python: 3.11.3
 * Git: 2.30.2
 * psycopg2: 2.9.6
 * psycopg2-binary: 2.9.6
 * phply: 1.2.6
 * ruamel.yaml: 0.17.31
 * tesserocr: 2.6.0
 * boto3: 1.26.146
 * zeep: 4.2.1
 * aeidon: 1.12
 * iniparse: 0.5
 * mysqlclient: 2.1.1
 * Mercurial: 6.4.3
 * git-svn: 2.30.2
 * git-review: 2.3.1
 * Redis server: 5.0.14
 * PostgreSQL server: 14.8
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: mail.gandi.net
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 5.15.0-72-generic (x86_64)

### Weblate deploy checks

```shell
System check identified some issues:

WARNINGS:
?: (security.W004) You have not set a value for the SECURE_HSTS_SECONDS setting. If your entire site is served only over SSL, you may want to consider setting a value and enabling HTTP Strict Transport Security. Be sure to read the documentation first; enabling HSTS carelessly can cause serious, irreversible problems.
?: (security.W008) Your SECURE_SSL_REDIRECT setting is not set to True. Unless your site should be available over both SSL and non-SSL connections, you may want to either set this setting True or configure a load balancer or reverse-proxy server to redirect all connections to HTTPS.
?: (security.W012) SESSION_COOKIE_SECURE is not set to True. Using a secure-only session cookie makes it more difficult for network traffic sniffers to hijack user sessions.
?: (security.W018) You should not have DEBUG set to True in deployment.

INFOS:
?: (weblate.I021) Error collection is not set up, it is highly recommended for production use
	HINT: https://docs.weblate.org/en/weblate-4.17/admin/install.html#collecting-errors
?: (weblate.I028) Backups are not configured, it is highly recommended for production use
	HINT: https://docs.weblate.org/en/weblate-4.17/admin/backup.html
?: (weblate.I031) New Weblate version is available, please upgrade to 4.18.
	HINT: https://docs.weblate.org/en/weblate-4.17/admin/upgrade.html

System check identified 7 issues (1 silenced).
```

**Comment by nijel** [comment]
Please upgrade to 4.18, this has been already fixed in 428f7114be3bc79be61bd9fc26cd0472de3c68fb.

**Comment by Rexogamer** [comment]
I did check the 4.18 changelog/recent issues before opening this but didn't
see it anywhere - however, thanks for letting me know; I can confirm that,
after upgrading, this issue has been resolved :3
